src/pixelator/graph/backends/implementations.py

```diff
@@ -6,12 +6,15 @@
 from __future__ import annotations
 
 import logging
-from typing import Dict, Iterable, List, Optional, Tuple
+import warnings
+from typing import Dict, Iterable, List, Optional, Tuple, Union
 
 import igraph
```

Comment:
To separate better, should we move this imports within the classes? Or the classes on different files?

Reply:
I don't think we should import inside the classes, since the recommendation in python is to import at the top of the file so that all imports are resolved at import time. But, we can split this into separate files for igraph and networkx. I suggest we do that in a separate PR though.